omegatrade-app/frontend/src/app/components/company/manage-company/manage-company.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, ViewChild } from '@angular/core';
+import { MatSort } from '@angular/material/sort';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatDialog } from '@angular/material/dialog';
+import { UpdateCompanyComponent } from '../update-company/update-company.component';
+import { RestService } from '../../../services/rest.service';
+import { SnackBarService } from '../../../services/snackbar.service';
+import { take } from "rxjs/operators";
+
+@Component({
+  selector: 'app-manage-company',
+  templateUrl: './manage-company.component.html',
+  styleUrls: ['./manage-company.component.css']
+})
+
+export class ManageCompanyComponent {
+  displayedColumns: string[] = ['companyName', 'companyShortCode', 'action'];
+  dataSource: MatTableDataSource<CompanyData>;
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  response: any;
+  loader: boolean = false;
+
+  constructor(private snackBarService: SnackBarService, private restService: RestService, public dialog: MatDialog) {
+    this.getCompanies();
+  }
+
+  getCompanies() {
+    this.loader = true;
+    this.restService.getData('companies/list')
+    .pipe(take(1))
+    .subscribe(response => {
+      this.response = response;
+      if (this.response && this.response.success) {
+        this.dataSource = new MatTableDataSource(this.response.data);
+        this.dataSource.paginator = this.paginator;
+        this.dataSource.sort = this.sort;
```

Comment:
Instead of the assigning these two properties here in the each companies list call, we should have it separately like
```
ngAfterViewInit() {
    this.dataSource.paginator = this.paginator;
    this.dataSource.sort = this.sort;
  }
  ```